lib/network/modules/LayoutEngine.js

```diff
@@ -1684,55 +1664,96 @@ class LayoutEngine {
     return findParent(parents, childB);
   }
 
+
   /**
-   * Abstract the getting of the position so we won't have to repeat the check for direction all the time
-   * @param {vis.Node} node
-   * @param {{x: Number, y: Number}} position
-   * @param {Number} level
-   * @param {boolean} [doNotUpdate=false]
+   * Set the strategy pattern for handling the coordinates given the current direction.
+   *
+   * The individual instances contain all the operations and data specific to a layout direction.
+   *
    * @private
    */
-  _setPositionForHierarchy(node, position, level, doNotUpdate = false) {
-    //console.log('_setPositionForHierarchy',node.id, position)
-    if (doNotUpdate !== true) {
-      this.hierarchical.addToOrdering(node, level);
-    }
+  setDirectionStrategy() {
+    var isVertical = (this.options.hierarchical.direction === 'UD'
+                   || this.options.hierarchical.direction === 'DU');
+
+    if(isVertical) {
+      this.direction = {
```

Comment:
Imo, it would be better to define a direction interface, and have two fleshed out classes that implement the direction interface. That way the code is self documenting.

Reply:
That's not a bad idea. Haven't actually used interfaces in javascript because there's no technical need for it.

Google is telling me there's no such thing, even for >= ES5. It would be a hackish kind of thing.

Reply:
I see.

Reply:
Maybe a typedef describing the expectations of a sort strategy and what it's expected to implement.

Reply:
Ah, here's a good example of an abstract base class that would achieve the same end: https://stackoverflow.com/questions/597769/how-do-i-create-an-abstract-base-class-in-javascript

Reply:
OK. I adapted the idea to es5 classes. See how you like it.